### PR TITLE
Dp 2480

### PR DESCRIPTION
**Pull request by shriharirathod**

### CODE REVIEW

File: src/hyc_asd_mgr.py [hunk 1]
```diff
@@ -246,6 +250,10 @@ def is_service_avaliable():
     out, err = ret.communicate()
     status = ret.returncode
     if status:
+        log.error("is_service_available failed!!!")
+        log.error("out: %s" %out)
```
Comment (sumit-k) [inline src/hyc_asd_mgr.py]: replace "out" to "stdout"

File: src/hyc_asd_mgr.py [hunk 2]
```diff
@@ -386,10 +396,29 @@ def run(self):
...
+
+        lease_duration = self.halib.get_health_lease()
+        while (True):
+            if (not is_service_up() and self.failure_retry > 0):
+                log.error("service not up!! Retry cnt: %s" %self.failure_retry)
+                self.failure_retry -= 1
+                time.sleep(lease_duration / 4)
+                continue
+
+            if (not is_service_avaliable() and self.failure_retry > 0):
```
Comment (sumit-k) [inline src/hyc_asd_mgr.py:409]: Test this scenario..

Comment (sumit-k) [inline src/hyc_asd_mgr.py:409]: test this variant.

**Review by sumit-k — APPROVED** [review]
Looks good

File: src/hyc_asd_mgr.py (hunk 1 above)
Comment (shriharirathod) [reply src/hyc_asd_mgr.py]: Done

File: src/hyc_asd_mgr.py (hunk 2 above)
Comment (shriharirathod) [reply src/hyc_asd_mgr.py:409]: Tested and changes working as expected.

**Comment by shriharirathod** [comment]
Test scenarios added on original JIRA: https://primaryio.atlassian.net/browse/DP-2480

**Pull request merged**